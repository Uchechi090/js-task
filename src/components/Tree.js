import React, { Component } from "react";
import PropTypes from 'prop-types';

import data from "../nodes";
import TreeNode from "./TreeNode";

class Tree extends Component {
    constructor(props) {
        super(props);

        this.state = {
            nodes: data
        };
    }

    getParentNodes = (rootNode) => {
        return rootNode.filter(node => node.hasOwnProperty("parent") === false);
    }

    getChildNodes = (parentNodeId) => {
        const { nodes } = this.state;
        return nodes.filter(node => node.parent === parentNodeId);
    }


    check = (id) => {
        document.getElementById(id).checked = true;
    }

    // uncheck = (id) => {
    //     document.getElementById(id).checked = false;
    // }

    toggleNode = (node) => {
        const { nodes } = this.state;
        nodes[node.id].isOpen = !node.isOpen;
        this.setState({ nodes });

        //this.check(node.id);
        // this.uncheck(node.id);
    }

    nodeSelect = node => {
        const { onSelect } = this.props;
        onSelect(node);
    }

    render() {
        //console.log(this.getParentNodes())
        //console.log(this.getChildNodes(0))
        const { nodes } = this.state;
        const parentNodes = this.getParentNodes(nodes);
        return (
            <div>
                {parentNodes.map(node => (
                    <TreeNode
                        key={node.id}
                        node={node}
                        getChildNodes={this.getChildNodes}
                        toggleNode={this.toggleNode}
                        nodeSelect={this.nodeSelect}
                    // level={0}
                    />
                ))}
            </div>
        )
    }

}

// Tree.propTypes = {
//     onSelect: PropTypes.func.isRequired,
// };

export default Tree;