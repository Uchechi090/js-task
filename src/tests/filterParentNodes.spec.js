import React from "react";
import { create } from "react-test-renderer";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import Tree from '../components/Tree';

describe("Tree class component", () => {
    test("it should make a snapshot test", () => {
        const tree = create(<Tree />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});

describe("Get parent nodes", () => {
    let wrapper;
    let mockSubmit;
    beforeEach(() => {
        mockSubmit = jest.fn();
        wrapper = shallow(<Tree submit={mockSubmit} />);
    })
    test("it should filter an array", () => {
        const input = [
            { id: 0, name: 'aaa' },
            { id: 1, name: 'bbb', parent: 0 },
            { id: 2, name: 'ccc', parent: 0 },
            { id: 3, name: 'ddd' },
            { id: 4, name: 'eee', parent: 3 },
            { id: 5, name: 'fff', parent: 3 },
            { id: 6, name: 'ggg', parent: 3 },
            { id: 7, name: 'hhh' },
            { id: 8, name: 'iii', parent: 7 },
            { id: 9, name: 'jjj', parent: 8 },
            { id: 10, name: 'kkk', parent: 9 },
            { id: 11, name: 'lll', parent: 9 },
            { id: 12, name: 'mmm', parent: 9 },
            { id: 13, name: 'nnn', parent: 8 },
            { id: 14, name: 'ooo', parent: 13 },
            { id: 15, name: 'ppp', parent: 13 },
            { id: 16, name: 'qqq', parent: 13 },
            { id: 17, name: 'rrr' },
            { id: 18, name: 'sss', parent: 17 },
            { id: 19, name: 'ttt', parent: 17 },
            { id: 20, name: 'uuu', parent: 19 },
            { id: 21, name: 'vvv', parent: 19 },
            { id: 22, name: 'www', parent: 17 },
            { id: 23, name: 'xxx', parent: 17 },
            { id: 24, name: 'yyy', parent: 23 },
            { id: 25, name: 'zzz', parent: 23 },
        ];
        const output = [{ id: 0, name: 'aaa' }, { id: 3, name: 'ddd' }, { id: 7, name: 'hhh' }, { id: 17, name: 'rrr' }];
        const tree = shallow(<Tree />);
        const getParentNodes = tree.find().getParentNodes()
        expect(getParentNodes(input)).toEqual(output);
    });
});