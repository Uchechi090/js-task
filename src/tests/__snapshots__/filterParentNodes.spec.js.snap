// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter function it should filter an array 1`] = `
<div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      aaa
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      ddd
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      hhh
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      rrr
    </span>
  </div>
</div>
`;

exports[`G it should make a snapshot test 1`] = `
<div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      aaa
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      ddd
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      hhh
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      rrr
    </span>
  </div>
</div>
`;

exports[`Get parent nodes it should filter  1`] = `
<div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      aaa
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      ddd
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      hhh
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      rrr
    </span>
  </div>
</div>
`;

exports[`Get parent nodes it should filter an array 1`] = `
<div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      aaa
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      ddd
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      hhh
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      rrr
    </span>
  </div>
</div>
`;

exports[`Get parent nodes it should make a snapshot test 1`] = `
<div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      aaa
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      ddd
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      hhh
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      rrr
    </span>
  </div>
</div>
`;

exports[`Tree class component it should filter an array 1`] = `
<div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      aaa
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      ddd
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      hhh
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      rrr
    </span>
  </div>
</div>
`;

exports[`Tree class component it should make  1`] = `
<div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      aaa
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      ddd
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      hhh
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      rrr
    </span>
  </div>
</div>
`;

exports[`Tree class component it should make a snapshot test 1`] = `
<div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      aaa
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      ddd
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      hhh
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      rrr
    </span>
  </div>
</div>
`;

exports[`Tree class it should filter an array 1`] = `
<div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      aaa
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      ddd
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      hhh
    </span>
  </div>
  <div
    className="sc-bdVaJa kePYVv"
  >
    <div
      className="sc-bwzfXH cczQhJ"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
    </div>
    <span>
      rrr
    </span>
  </div>
</div>
`;
